core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -33,17 +33,31 @@
 public class CachingCatalog implements Catalog {
 
   public static Catalog wrap(Catalog catalog) {
-    return new CachingCatalog(catalog);
+    return wrap(catalog, true);
+  }
+
+  public static Catalog wrap(Catalog catalog, Boolean caseSensitive) {
+    return new CachingCatalog(catalog, caseSensitive);
   }
 
   private final Cache<TableIdentifier, Table> tableCache = Caffeine.newBuilder()
       .softValues()
       .expireAfterAccess(1, TimeUnit.MINUTES)
       .build();
   private final Catalog catalog;
+  private final Boolean caseSensitive;
 
-  private CachingCatalog(Catalog catalog) {
+  private CachingCatalog(Catalog catalog, Boolean caseSensitive) {
     this.catalog = catalog;
+    this.caseSensitive = caseSensitive;
+  }
+
+  private TableIdentifier formatIdentifier(TableIdentifier tableIdentifier) {
```

Comment:
I don't think that "format" is the right verb. How about "canonicalize"? I know that sounds more like jargon, but the method is really converting to a canonical version, not formatting.

Reply:
Fine with `canonicalizeIdentifier`.

The `format` prefix comes from Spark's SessionCatalog's `formatTableName` and `formatDatabaseName`.